skimage/_shared/testing.py

```diff
@@ -227,14 +227,21 @@ def wrapper(func):
         @functools.wraps(func)
         def inner(*args, **kwargs):
             threads = []
-            for i in range(num_threads):
+            for i in range(num_threads - 1):
                 thread = threading.Thread(target=func, args=args, kwargs=kwargs)
                 threads.append(thread)
             for thread in threads:
                 thread.start()
+
+            result = func(*args, **kwargs)
+
             for thread in threads:
                 thread.join()
+
+            return result
```

Comment:
with this, is the Notes section of the docstring still true? 
"This decorator does not pass the return value of the decorated function."